File: python_backend/game_logic/threatArea/get_range_instance.py
from game_logic.test_objects.get_standard_piece_defs import get_standard_piece_defs


def get_range_instance(range_defs, piece_name, color, range_type):
    """get range of range_type of the piece named piece_name of enemy color"""
    return range_defs[piece_name][color][range_type]


if __name__ == "__main__":
    range_defs = get_standard_piece_defs()
    print(get_range_instance(range_defs, "Rook", "W", "spans"))
    print(get_range_instance(range_defs, "Rook", "W", "offsets"))
    print(get_range_instance(range_defs, "Queen", "B", "spans"))
    print(get_range_instance(range_defs, "Queen", "B", "offsets"))

File: node.js/threatArea/getRangeInstance.js
var  getStandardPieceDefs  = require("gameLogic/testObjects/getStandardPieceDefs");


function getRangeInstance(rangefunctions, pieceName, color, rangeType) {
    /*get range of rangeType of the piece named pieceName of enemy color**/
    return rangeDefs[pieceName][color][rangeType]
}

module.exports = getRangeInstance;

if (require.main === module) {
    let rangeDefs = getStandardPieceDefs()
    console.log(getRangeInstance(rangeDefs, "Rook", "W", "spans"))
    console.log(getRangeInstance(rangeDefs, "Rook", "W", "offsets"))
    console.log(getRangeInstance(rangeDefs, "Queen", "B", "spans"))
    console.log(getRangeInstance(rangeDefs, "Queen", "B", "offsets"))
}